### DISCUSSION TITLE
Sof-Firmware on FreeBSD

### DISCUSSION THREAD

**Discussion by jb1277976** [thesofproject/sof#9506] [General]
Will any of the BSD's ever get sof-firmware ?

I'm pulling my hair out over here nothing is working..

I have got sound working on every single distro i have tired and it just works.. right now I'm into freebsd and would like internal speakers and microphone to work natively..

Any ideas if it will ever happen?

Thanks

Joe

**Comment by plbossart** [comment]
The licensing for all the SOF drivers is permissive on purpose to allow for ports to non-Linux/GPL environments. There is already a Windows driver based on SOF, so it's possible. That said it's a matter of time and resources that will most likely require community engagement.

There are still two technical problems
- there's a fairly strong dependency on ASoC/DAPM/DPCM for power management, this environment does not have any equivalent in BSDs to the best of my knowledge.
- Things are becoming more complex with the transition from HDaudio (first introduced in early 2000s) to the SoundWire bus, that would require even more plumbing to support basic bus functionality and codec drivers. At some point this will require active support from 3rd parties such as Realtek/Cirrus Logic.

**Reply by lgirdwood** [reply]
Further to @plbossart answer, you may want to take the Linux driver permissive parts and hard code a topology in FreeBSD driver for initial integration. That would get audio working for simple use cases, the Windows driver may do this and is probably a good reference too.

**Comment by jb1277976** [comment]
Thanks all..

This won't be a problem since i ordered a Lenovo ThinkPad L440 14" Laptop Intel Core i3 8GB RAM 500GB HDD Windows 10 Pro which everything works on freeBSD.. i'm wondering if i should close this issue in case somebody that has the same issue wants to know why sound dosen't work on freebsd that requires sof-firmware ?

**Reply by lgirdwood** [reply]
I would keep this open as legacy HDA has being deprecated within the industry and being replaced by Soundwire with optional (depending on vendor) DSP support for offloading. Its probably worthwhile to make sure FreeBSD gets good soundwire and DSP support as audio may not work in the future without it on most devices.

**Reply by jb1277976** [reply]
> I would keep this open as legacy HDA has being deprecated within the industry and being replaced by Soundwire with optional (depending on vendor) DSP support for offloading. Its probably worthwhile to make sure FreeBSD gets good soundwire and DSP support as audio may not work in the future without it on most devices.

Thanks..

I've gotten another laptop as before l440 and everything works fine under openbsd.. will keep my chromebook with freebsd for quite a while longer till everything works in the future..

Thanks

Joe
